packages/react-app/src/helpers/Transactor.js

```diff
@@ -115,7 +115,7 @@ export default function Transactor(providerOrSigner, gasPrice, etherscan) {
         console.log("Transaction Error:", e.message);
         notification.error({
           message: "Transaction Error",
-          description: e.message,
+          description: e.data ? e.data.message : e.message,
```

Comment:
This logic works for Metamask, but I think this may need additional handling for the burnerSigner - I see this: 
![image](https://user-images.githubusercontent.com/9612972/123489301-8cc98980-d609-11eb-9c84-4295c83d670c.png)

Essentially the error thrown seems to be different. I got it showing up nicely with the burner by looking for `e.reason`, i.e.
```
notification.error({
          message: "Transaction Error",
          description: e.data ? e.data.message : e.reason ? e.reason : e.message,
        });```
though that's a bit gnarly!
I tried with WalletConnect via Metamask mobile, which didn't actually throw at all, so maybe there is something required on that side.

Reply:
Thanks @azf20 while I agree it looks bad, this issue is not something that has been introduced by this PR. If on the master branch you do the same you will see the same result if the function reverts. Check this (on master branch):
```
  function setPurpose(string memory newPurpose) public {
    require (bytes(newPurpose).length > 0, "Purpose is empty!");
    purpose = newPurpose;
    console.log(msg.sender,"set purpose to",purpose);
    emit SetPurpose(msg.sender, purpose);
  }
```

**With BurnerWallet:**
<img width="1247" alt="Screen Shot 2021-06-25 at 9 08 44 PM" src="https://user-images.githubusercontent.com/526558/123495383-e1650880-d5f9-11eb-9176-bca6a5abfde6.png">

**With Metamask:**
<img width="849" alt="Screen Shot 2021-06-25 at 9 08 32 PM" src="https://user-images.githubusercontent.com/526558/123495405-ef1a8e00-d5f9-11eb-8db1-57a5fd709323.png">

So, I wouldn't stop this PR because of this, since we are still getting the same behavior, but possible open another issue to get it fixed. Although, I've been looking at this for a while and can't find a way to get that error correctly parsed, which seems to be the issue (when using the burnerWallet the error is not parsed correctly by ethers.js?)

Reply:
Sure I think the snippet I shared in the comment parses the error correctly for the BurnerSigner:
```
notification.error({
          message: "Transaction Error",
          description: e.data ? e.data.message : e.reason ? e.reason : e.message,
        });
```
I'll push to this branch and you can verify?